### PR TITLE
use es2015 and react 0.14

### PR DESCRIPTION
**Pull request by chentsulin**
#58, #59

### CODE REVIEW

**Comment by elijahsmith** [comment]
it's exciting to see the recent updates to this project; we're relying on it for a fairly significant application that's releasing in a few months and were on the verge of moving away from react-ace.

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# http://editorconfig.org
```
Comment (elijahsmith) [inline .editorconfig]: this seems unnecessary for this PR...

File: example/.babelrc [hunk 2]
```diff
@@ -1,17 +1,17 @@
...
-			"extra": {
-				"react-transform": [{
-					"target":  "react-transform-hmr",
-					"imports": ["react"],
-					"locals":  ["module"]
-				}]
-			}
-		}
-	}
+  "stage": 2,
```
Comment (elijahsmith) [inline example/.babelrc]: I'm not sure about this change either, as it's purely whitespace preference...

File: src/ace.jsx [hunk 3]
```diff
@@ -1,152 +1,105 @@
...
-    this.editor = ace.edit(this.props.name);
-    if (this.props.onBeforeLoad) {
-      this.props.onBeforeLoad(ace);
+import ace from 'brace';
+import React, { Component, PropTypes } from 'react';
+
+export default class ReactAce extends Component {
+  constructor(props) {
+    super(props);
+    [
```
Comment (elijahsmith) [inline src/ace.jsx]: why is this (binding these methods to ```this```) necessary?

File: .editorconfig (hunk 1 above)
Comment (chentsulin) [inline .editorconfig]: I want this file but I can split it to another PR.

File: example/.babelrc (hunk 2 above)
Comment (chentsulin) [inline example/.babelrc]: new config format
https://github.com/gaearon/babel-plugin-react-transform/releases/tag/v1.1.0

File: src/ace.jsx (hunk 3 above)
Comment (chentsulin) [inline src/ace.jsx]: ES6 class will not auto bind this for us, so we should bind this at constructor or for every call site.

**Comment by soulmachine** [comment]
This PR looks great!

**Comment by securingsincity** [comment]
thanks @chentsulin  looks great

**Pull request merged**